app/Nova/User.php

```diff
@@ -809,4 +813,36 @@ private static function adminCanRun(NovaRequest $request): bool
     {
         return $request->user()->hasRole('admin');
     }
+
+    /**
+     * Helper method: Graduation semester is a 6-digit code by default.
+     * Prepares semester data for display, returning empty if no string/invalid
+     * string is entered.
+     */
+    private static function parseGradSemester(?string $sem): string
+    {
+        if ($sem === null) {
+            return '';
+        }
+        if (preg_match('/^[0-9]{4}0[258]$/', $sem) === 0 || preg_match('/^[0-9]{4}0[258]$/', $sem) === false) {
+            return '';
+        }
```

Comment:
- PHP shouldn't evaluate later parts of an OR conditional if the first is true, so you can merge this `if` statement with the previous null check
- I would pull out the `preg_match` here since it's called identically twice; it'll make the condition more compact